### PR TITLE
[LPT] Slice transformation

### PR DESCRIPTION
**Pull request by eshoguli**
### Details:
 - *Slice transformation (per-tensor quantization)*

### Tickets:
 - *CVS-149909*

### CODE REVIEW

**Review by v-Golubev — COMMENTED** [review]
Please also add `Slice` op to `MarkupPrecisions::isPrecisionPreserved` and `MarkupPrecisions::isSupported` lists

File: src/common/low_precision_transformations/src/slice.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    this->register_matcher(m, callback);
+}
+
+bool SliceTransformation::transform(TransformationContext& context, ov::pass::pattern::Matcher& m) {
+    if (!SliceTransformation::canBeTransformed(context, m.get_match_root())) {
+        return false;
+    }
+
+    const auto strided_slice = NetworkHelper::separateInStandaloneBranch(m.get_match_root(), defaultPrecisions);
+    auto dequantization = NetworkHelper::getDequantization(strided_slice, defaultPrecisions);
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/common/low_precision_transformations/src/slice.cpp]: removed, thanks

File: src/common/low_precision_transformations/src/slice.cpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    if (!ov::is_type<ov::opset8::Slice>(operation)) {
+        return false;
+    }
+
+    const auto dequantization = NetworkHelper::getDequantization(operation);
+    if (dequantization.empty()) {
+        return false;
+    }
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/common/low_precision_transformations/src/slice.cpp:60]: will discuss offline

File: src/tests/functional/plugin/shared/src/low_precision_transformations/slice_transformation.cpp [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    os << "{ ";
+    for (size_t i = 0; i < values.size(); ++i) {
+        os << values[i];
+        if (i != (values.size() - 1ul)) {
+            os << ", ";
+        }
+    }
+    os << " }";
+    return os;
+}
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/tests/functional/plugin/shared/src/low_precision_transformations/slice_transformation.cpp]: done, thanks

File: src/plugins/intel_cpu/tests/functional/shared_tests_instances/low_precision_transformations/slice_transformation.cpp
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/plugins/intel_cpu/tests/functional/shared_tests_instances/low_precision_transformations/slice_transformation.cpp:1]: was added, thanks

File: src/plugins/intel_gpu/tests/functional/shared_tests_instances/low_precision_transformations/slice_transformation.cpp [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        { 2 }, // axes
+        "f32"
+    },
+};
+
+INSTANTIATE_TEST_SUITE_P(smoke_LPT, SliceTransformation,
+    ::testing::Combine(
+        ::testing::ValuesIn(netPrecisions),
+        ::testing::Values(ov::PartialShape({ 1, 3, 24, 24 })),
+        ::testing::Values(ov::test::utils::DEVICE_CPU),
```
Comment (v-Golubev) [inline src/plugins/intel_gpu/tests/functional/shared_tests_instances/low_precision_transformations/slice_transformation.cpp:59]:
```suggestion
        ::testing::Values(ov::test::utils::DEVICE_GPU),
```

**Review by v-Golubev — COMMENTED** [review]
Markup passes still need to be updated ([details](https://github.com/openvinotoolkit/openvino/pull/26609#pullrequestreview-2307267852))

File: src/plugins/intel_gpu/tests/functional/shared_tests_instances/low_precision_transformations/slice_transformation.cpp (hunk 4 above)
Comment (eshoguli) [reply src/plugins/intel_gpu/tests/functional/shared_tests_instances/low_precision_transformations/slice_transformation.cpp]: done, thanks

**Pull request merged**